pkg/ingester/client/circuitbreaker.go

```diff
@@ -69,7 +69,13 @@ func NewCircuitBreaker(inst ring.InstanceDesc, cfg CircuitBreakerConfig, metrics
 			transitionHalfOpen.Inc()
 			level.Info(logger).Log("msg", "circuit breaker is half-open", "ingester", inst.Id, "previous", event.OldState, "current", event.NewState)
 		}).
-		HandleIf(func(r any, err error) bool { return isFailure(err) }).
+		HandleIf(func(r any, err error) bool {
+			isFail := isFailure(err)
+			if isFail {
+				level.Warn(logger).Log("msg", "circuit breaker detected a failing execution", "ingester", inst.Id, "err", err)
```

Comment:
If you're making changes to these log messages, we should include the instance address here too.